File: salimmo-parent/property-service/src/main/java/org/propertyservice/dto/PropertySurfaceDto.java
package org.propertyservice.dto;

import lombok.*;

@AllArgsConstructor
@NoArgsConstructor
@Getter
@Setter
@ToString
public class PropertySurfaceDto {
    private Long id;
    //@JsonIgnoreProperties("propertySurface")
    private PropertyDto property;
    private Double livingArea;
    private Double groundSurface;
    private Double usefulSurface;
    private Double roofSpace;
    private Double squareArea;
    private Double gardenArea;
    private Double terraceArea;
    private Double balconyArea;
    private Double weightedArea;
    private Double ancillaryArea;
    private Double buildingArea;
}

File: salimmo-frontend/src/app/models/propertysurface/propertySurfaceDto.ts
import {PropertyDto} from "../property/propertyDto";

export class PropertySurfaceDto {
  id!:number;
  property!:PropertyDto;
  livingArea!:number;
  groundSurface!:number;
  usefulSurface!:number;
  roofSpace!:number;
  squareArea!:number;
  gardenArea!:number;
  terraceArea!:number;
  balconyArea!:number;
  weightedArea!:number;
  ancillaryArea!:number;
  buildingArea!:number;
}
